Thesis/StepIndexedRelBigStepTypedAnfIlcCorrect.agda

```diff
@@ -0,0 +1,332 @@
+-- Step-indexed logical relations based on relational big-step semantics
```

Comment:
This development is what I did today on the whiteboard (modulo constants).
The only lemma I needed was monotonicity of validity (plus arithmetic on step-indexes, which Coq supports better).

This definition of validity doesn't agree with ⊕, but it allows function bodies to change. Hence, this development also describes functions that change without replacement changes.
Remember that logical relation define a semantic model (with some behavior of denotational properties), even if it's based on syntax. Hence, I'll conjecture that "essentially" the same relation should also allow for functions changes that keep the body + replacement closures—that's a different syntax, but should work with "essentially" the same relation (and in particular, the same relation on computations).

And it should be very easy to forbid changes to the body.